samples/bluetooth/peripheral_uart_nrf53/common/bt_ser.c

```diff
@@ -0,0 +1,612 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cbor.h>
+#include <errno.h>
+#include <kernel.h>
+#include <zephyr/types.h>
+
+#include <bluetooth/addr.h>
+
+#if defined(NRF5340_XXAA_NETWORK)
+#include <bluetooth/services/nus.h>
+#endif
+
+#include "bt_ser.h"
+#include "rpmsg.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(bt_ser);
+
+#define BT_CMD_GATT_NUS_SEND_ARRAY_SIZE 0x03
+#define BT_EVT_SEND_ARRAY_SIZE 0x05
+#define BT_RSP_RESULT_ARRAY_SIZE 0x02
+#define BT_NUS_INIT_CMD_ARRAY_SIZE 0x02
+
+#define BT_CMD_BT_NUS_INIT 0x01
+#define BT_CMD_GATT_NUS_SEND 0x02
+
+#define BT_EVENT_CONNECTED 0x01
+#define BT_EVENT_DISCONNECTED 0x02
+#define BT_EVENT_DATA_RECEIVED 0x03
+
+#define BT_TYPE_CMD 0x01
+#define BT_TYPE_EVENT 0x02
+#define BT_TYPE_RSP 0x03
+
+#define APP_TO_NETWORK_MAX_PKT_SIZE CONFIG_APP_TO_NETWORK_MAX_PACKET_SIZE
+#define NETWORK_TO_APP_MAX_PKT_SIZE CONFIG_NETWORK_TO_APP_MAX_PACKET_SIZE
+
+#define APP_WAIT_FOR_RSP_TIME CONFIG_APP_WAIT_FOR_RESPONSE_TIME
+
+K_SEM_DEFINE(rsp_sem, 0, 1);
+
+typedef int (*cmd_rsp_handler_t)(CborValue *it);
+typedef int (*cmd_handler_t)(CborValue *it, CborEncoder *encoder);
+
+struct cmd_list {
+	u8_t cmd;
+	cmd_handler_t func;
+};
+
+static const struct bt_nus_cb *bt_cb;
+static int return_value;
+static cmd_rsp_handler_t rsp_handler;
+
+static int response_code_encode(CborEncoder *encoder, int rsp)
+{
+	int err;
+	CborEncoder array;
+
+	err = cbor_encoder_create_array(encoder, &array,
+					BT_RSP_RESULT_ARRAY_SIZE);
+
+	err = cbor_encode_simple_value(&array, BT_TYPE_RSP);
+	if (err) {
+		return -EFAULT;
+	}
+
+	if (cbor_encode_int(&array, rsp)) {
+		return -EFAULT;
+	}
+
+	err = cbor_encoder_close_container(encoder, &array);
+	if (err) {
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+#if defined(NRF5340_XXAA_NETWORK)
+__weak void bt_ready(int err)
+{
+	LOG_INF("Bluetooth initialized, err %d.", err);
+}
+
+static int bt_cmd_bt_nus_init(CborValue *it, CborEncoder *encoder)
+{
+	int err;
+
+	err = bt_enable(bt_ready);
+	if (err) {
+		LOG_ERR("Failed to enable Bluetooth.");
+	}
+
+	return err;
+}
+
+static int bt_cmd_gatt_nus_exec(CborValue *it, CborEncoder *encoder)
```

Comment:
worth to add a note that this function has to be of `cmd_handler_t` type

Reply:
Ok. I added brief for it and also for `bt_cmd_bt_nus_init`

Reply:
you must have forgot to push these changes here - I can't see them anywhere

Reply:
Yes. I did only rebese and I forgot about `commit --amend`